bika/lims/catalog/indexers/analysisrequest.py

```diff
@@ -28,3 +28,41 @@ def assigned_state(instance):
             return 'unassigned'
 
     return 'assigned'
+
+
+@indexer(IAnalysisRequest)
+def listing_searchable_text(instance):
+    """ Indexes values of desired fields for searches in listing view. All the
+    field names added to 'plain_text_fields' will be available to search by
+    wildcards.
+    Please choose the searchable fields carefully and add only fields that
+    can be useful to search by. For example, there is no need to add 'SampleId'
+    since 'getId' of AR already contains that value. Nor 'ClientTitle' because
+    AR's are/can be filtered by client in Clients' 'AR Listing View'
+    :return: values of the fields defined as a string
+    """
+    entries = []
+    plain_text_fields = ('getId', 'getContactFullName', 'getSampleTypeTitle',
+                         'getSamplePointTitle', 'getCreatorFullName',
+                         'getProfilesTitle', 'getStorageLocationTitle',
+                         'getClientOrderNumber', 'getClientReference',
+                         'getClientSampleID', 'getTemplateTitle', )
```

Comment:
Better go here through all available columns (see: https://github.com/senaite/senaite.core/blob/master/bika/lims/browser/bika_listing.py#L1163) and call a (recursive) "process_value" function to handle lists/dictionaries and other nested types as well, e.g. like here: https://github.com/senaite/senaite.publisher/blob/master/src/senaite/publisher/reportmodel.py#L257

Reply:
Yes, the index get the instance as the first argument and is therefore capable to return attributes from the object to the catalog index as well.

However, it is also possible to fetch the assigned catalog of the portal_type from the archetypes_tool and have access to all the metadata columns.

So it might be indeed an advantage to have access the attributes, but I would rather use that to augment the already existing metadata columns.

A more flexible approach makes this whole effort you invest at the moment into this much more sustainable w/o further need for code changes in the core...